Mathlib/MeasureTheory/Integral/RieszMarkovKakutani.lean

```diff
@@ -245,9 +256,145 @@ lemma rieszContentAux_union {K₁ K₂ : TopologicalSpace.Compacts X}
+          intro x hx
+          simp only [mem_preimage, mem_Ici] at hx
+          simp only [mem_support]
+          apply ne_of_gt
+          rw [Real.toNNReal_inv] at hx
+          exact (lt_of_lt_of_le
+            (inv_pos_of_pos (lt_trans zero_lt_one (Real.one_lt_toNNReal.mpr hα))) hx)
+      set hb' := hb ⟨K', hK'cp⟩
+      simp only [Compacts.coe_mk, le_iInf_iff] at hb'
+      have hbK' : b ≤ rieszContent Λ ⟨K', hK'cp⟩ := hb' hKK'
+      rw [ENNReal.le_coe_iff] at hbK'
+      obtain ⟨p, hp⟩ := hbK'
+      simp only [hp.1, ENNReal.toNNReal_coe, NNReal.val_eq_coe, map_smul, smul_eq_mul,
+        NNReal.coe_mul, Real.coe_toNNReal', ge_iff_le]
+      apply le_trans (NNReal.GCongr.toReal_le_toReal hp.2)
+      simp only [rieszContent]
+      rw [rieszContentAux, ← Real.coe_toNNReal (α ⊔ 0) (le_max_right α 0), ← NNReal.coe_mul,
+        NNReal.coe_le_coe]
+      apply csInf_le
+      · simp only [OrderBot.bddBelow]
+      · simp only [mem_image, mem_setOf_eq]
+        use α.toNNReal • f
+        refine ⟨?_, ?_⟩
+        · intro x hx
+          simp only [CompactlySupportedContinuousMap.coe_smul, Pi.smul_apply, smul_eq_mul,
+            ← NNReal.coe_le_coe, NNReal.coe_one, NNReal.coe_mul, Real.coe_toNNReal']
+          rw [← (left_eq_sup.mpr <| le_of_lt (lt_of_le_of_lt zero_le_one hα)), mul_comm]
+          apply (inv_le_iff_one_le_mul₀ (lt_trans zero_lt_one hα)).mp
+          rw [← Set.mem_Ici]
+          simp only [mem_Ici, ge_iff_le]
+          exact Real.toNNReal_le_iff_le_coe.mp hx
+        · simp only [map_smul, smul_eq_mul, mul_eq_mul_right_iff]
+          left
+          rw [Real.toNNReal_eq_toNNReal_iff (le_of_lt (lt_of_le_of_lt zero_le_one hα))
+            (le_max_right α 0), left_eq_sup]
+          exact le_of_lt (lt_of_le_of_lt zero_le_one hα)
+    · exact zero_le (Λ f)
+
+end RieszContentRegular
+
+section RieszMeasure
+
+variable [T2Space X] [LocallyCompactSpace X] [MeasurableSpace X] [BorelSpace X]
+
+/-- `rieszContent` is promoted to a measure. -/
+def rieszMeasure := (rieszContent Λ).measure
+
+lemma leRieszMeasure_Compacts {f : C_c(X, ℝ≥0)} (hf : ∀ (x : X), f x ≤ 1) {K : Compacts X}
+    (h : tsupport f ⊆ K) : ENNReal.ofNNReal (Λ f) ≤ (rieszMeasure Λ) K := by
+  simp only [rieszMeasure, Content.measure_eq_content_of_regular (rieszContent Λ)
+    (contentRegular_rieszContent Λ)]
+  simp only [rieszContent, ENNReal.ofReal_coe_nnreal, ENNReal.coe_le_coe]
+  apply le_iff_forall_pos_le_add.mpr
+  intro ε hε
+  obtain ⟨g, hg⟩ := exists_lt_rieszContentAux_add_pos Λ K hε
+  apply le_of_lt (lt_of_le_of_lt _ hg.2)
+  apply Λ_mono Λ
+  intro x
+  simp only [ContinuousMap.toFun_eq_coe, CompactlySupportedContinuousMap.coe_toContinuousMap]
+  by_cases hx : x ∈ tsupport f
+  · exact le_trans (hf x) (hg.1 x (Set.mem_of_subset_of_mem h hx))
+  · rw [image_eq_zero_of_nmem_tsupport hx]
+    exact zero_le (g x)
+
+lemma leRieszMeasure_Opens {f : C_c(X, ℝ≥0)} (hf : ∀ (x : X), f x ≤ 1) {V : Opens X}
```

Comment:
Why is this about `V : Opens X` rather than `V : Set X` + `IsOpen V`?

Reply:
I did so because in #12290 I defined `V` which takes value in `Opens`. Should I change?

Reply:
Yeah, more syntactically general lemmas are better, unless you think this lemma will only ever be used for `V : Opens X`, but I don't really see why that would be the case myself

Reply:
I see. done and changed the names of the lemmas